canonical-web-and-design/canonicalwebteam.http
# Use CacheControl module for better caching

## DONE
Implements 1 part of https://github.com/ubuntudesign/base-squad/issues/119
Fixes #5, #6 

## QA
Here is a test script to make sure that the code works.
This needs a Redis server running at the standard config locally.
```python
import time
import redis
from cachecontrol.caches.file_cache import FileCache
from cachecontrol.caches.redis_cache import RedisCache
from canonicalwebteam.http import CachedSession
from datetime import datetime


def create_redis_connection_pool(
    redis_port="6379", redis_host="localhost", redis_db=0
):
    return redis.ConnectionPool(host=redis_host, port=redis_port, db=redis_db)


pool = create_redis_connection_pool()
sess = CachedSession(redis_cache_pool=pool, fallback_cache_duration=10)
print(sess)
for i in range(9):
    print("start")
    print(datetime.now())
    response = sess.get("https://ehlers.berlin")

    print("is response from cache")
    print(response.from_cache)
    print(datetime.now())
    print(response.headers)
    time.sleep(10)
```
